### PR TITLE
junos_facts refactor to work with resources modules

### PR DESCRIPTION
**Pull request by ganeshrn**
##### SUMMARY

*  Refactor junos_facts module to work with
   network resource module.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

junos_facts
module_utils/network/common/facts/facts.py
module_utils/network/junos/argspec/facts/facts.py
module_utils/network/junos/facts/legacy/base.py

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: lib/ansible/module_utils/network/junos/argspec/facts/facts.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    """ The arg spec for the junos facts module
+    """
+
+    def __init__(self, **kwargs):
+        pass
+
+    argument_spec = {
+        'gather_subset': dict(default=['!config'], type='list'),
+        'config_format': dict(default='text', choices=['xml', 'text', 'set', 'json']),
+        'gather_network_resources': dict(choices=CHIOCES, type='list'),
```
Comment (samccann) [inline lib/ansible/module_utils/network/junos/argspec/facts/facts.py]: is this really correct that CHIOCES is misspelled here?

File: lib/ansible/modules/network/junos/junos_facts.py [hunk 2]
```diff
@@ -86,316 +101,28 @@
...
-            ansible_facts.update(OFacts(module).populate())
-        else:
-            warnings += ['junos-eznc is required to gather old style facts but does not appear to be installed. '
-                         'It can be installed using `pip  install junos-eznc`']
-        facts['gather_subset'].append('ofacts')
-
-    for key in runable_subsets:
-        instances.append(FACT_SUBSETS[key](module))
+    warnings = ['default value for `gather_subset` '
+                'will be changed to `min` from `!config` v2.11 onwards']
```
Comment (samccann) [inline lib/ansible/modules/network/junos/junos_facts.py:120]: `min` isn't listed currently as one of the possible values for `gather_subset`. Is that going to be added in 2.11?

File: lib/ansible/module_utils/network/junos/argspec/facts/facts.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+#
+# -*- coding: utf-8 -*-
+# Copyright 2019 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+"""
+The arg spec for the junos facts module.
+"""
+CHIOCES = [
```
Comment (samccann) [inline lib/ansible/module_utils/network/junos/argspec/facts/facts.py]:
```suggestion
CHOICES = [
```

File: lib/ansible/module_utils/network/junos/argspec/facts/facts.py (hunk 1 above)
Comment (samccann) [inline lib/ansible/module_utils/network/junos/argspec/facts/facts.py]:
```suggestion
        'gather_network_resources': dict(choices=CHOICES, type='list'),
```

File: lib/ansible/modules/network/junos/junos_facts.py (hunk 2 above)
Comment (ganeshrn) [reply lib/ansible/modules/network/junos/junos_facts.py:120]: Yes that's right

File: lib/ansible/module_utils/network/common/facts/facts.py [hunk 4]
```diff
@@ -23,7 +23,7 @@ def __init__(self, module):
         self._connection = get_resource_connection(module)
 
         self.ansible_facts = {'ansible_network_resources': {}}
-        self.ansible_facts['ansible_gather_network_resources'] = list()
+        self.ansible_facts['ansible_net_gather_network_resources'] = list()
```
Comment (danielmellado) [inline lib/ansible/module_utils/network/common/facts/facts.py:26]: what's the point on changing this? Will this just be part of another repo?

**Review by danielmellado — COMMENTED** [review]
Just some questions there, overall I thin the wrapper for lxml ET comes really handy. Thanks ganesh!

File: lib/ansible/module_utils/network/common/facts/facts.py (hunk 4 above)
Comment (ganeshrn) [reply lib/ansible/module_utils/network/common/facts/facts.py:26]: It is a typo, `ansible_net_*` is added in facts key as a convention. It is similar to `ansible_net_gather_facts`

**Comment by danielmellado** [comment]
recheck

**Pull request merged**